qmplay2: 25.01.19 -> 25.06.27

ProxyVT opened the pull request:
<!--
^ Please summarise the changes you have done and explain why they are necessary here ^

For package updates please link to a changelog or describe changes, this helps your fellow maintainers discover breaking updates.
For new packages please briefly describe the package or provide a link to its homepage.
-->
https://github.com/zaps166/QMPlay2/releases/tag/25.06.11
https://github.com/zaps166/QMPlay2/releases/tag/25.06.27

## Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform:
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- Tested, as applicable:
  - [ ] [NixOS tests] in [nixos/tests].
  - [ ] [Package tests] at `passthru.tests`.
  - [ ] Tests in [lib/tests] or [pkgs/test] for functions and "core" functionality.
- [x] Ran `nixpkgs-review` on this PR. See [nixpkgs-review usage].
- [x] Tested basic functionality of all binary files, usually in `./result/bin/`.
- Nixpkgs Release Notes
  - [ ] Package update: when the change is major or breaking.
- NixOS Release Notes
  - [ ] Module addition: when adding a new NixOS module.
  - [ ] Module update: when the change is significant.
- [x] Fits [CONTRIBUTING.md], [pkgs/README.md], [maintainers/README.md] and other READMEs.

[NixOS tests]: https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests
[Package tests]: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests
[nixpkgs-review usage]: https://github.com/Mic92/nixpkgs-review#usage

[CONTRIBUTING.md]: https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md
[lib/tests]: https://github.com/NixOS/nixpkgs/blob/master/lib/tests
[maintainers/README.md]: https://github.com/NixOS/nixpkgs/blob/master/maintainers/README.md
[nixos/tests]: https://github.com/NixOS/nixpkgs/blob/master/nixos/tests
[pkgs/README.md]: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md
[pkgs/test]: https://github.com/NixOS/nixpkgs/blob/master/pkgs/test

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

ProxyVT committed:
qmplay2: 25.01.19 -> 25.06.27

Fixes #380869

ProxyVT committed:
qmplay2: remove substituteInPlace

Fixes build error: substituteStream() in derivation qmplay2-qt6-25.06.27: ERROR: pattern getSubmitInfo\(\) doesn't match anything in file 'src/qmplay2/vulkan/VulkanWindow.cpp'

ProxyVT committed:
qmplay2: remove fetchSubmodules

We are already fetching submodules directly (https://github.com/NixOS/nixpkgs/commit/365e7e5d07f3b2962ab87ed28fcf74204577a712); there is no need to use fetchSubmodules.

pbsds commented:
```
error: hash mismatch in fixed-output derivation '/nix/store/pgvn1pflsjrxpxhaprrvxj3h72j5b7jc-source.drv':
        likely URL: https://github.com/zaps166/QMPlay2/archive/refs/tags/25.06.27.tar.gz
         specified: sha256-ouP9E9nk8T0EZr9D354hc/QKpyv7h3xpY1UCbo8Kw4k=
            got:    sha256-+kDaRKwXOHnHje1RntC9y9xiTaMzs8SGMLVoJ+6IDNk=
```

ProxyVT committed:
qmplay2: pin vulkan-headers dependency to specific version

This prevents compilation issues with QmVk library, as it is hardcoded for a specific version of Vulkan-Headers (the package repo uses 1.3.277). I chose version 1.3.300 as it is the latest version that doesn't cause build failures due to changes in the Vulkan SDK that are incompatible with QmVk.

Additionally, version control will help with debugging potential problems that may arise from future updates.
